FreeRTOS-Plus/Source/FreeRTOS-Plus-TCP/FreeRTOS_DNS.c

```diff
@@ -828,11 +828,11 @@ TickType_t uxWriteTimeOut_ticks = ipconfigDNS_SEND_BLOCK_TIME_TICKS;
 /*-----------------------------------------------------------*/
 
 _static size_t prvCreateDNSMessage( uint8_t *pucUDPPayloadBuffer,
-								   const char *pcHostName,
-								   TickType_t uxIdentifier )
+									const char *pcHostName,
+									TickType_t uxIdentifier )
 {
 DNSMessage_t *pxDNSMessageHeader;
-uint8_t *pucStart, *pucByte;
+size_t uxStart, uxIndex;
```

Comment:
While we are doing these changes, shall we also add a `configASSERT( pucUDPPayloadBuffer != NULL )`? Since that is directly used with the `memcpy` on line 849.
I would also like if we have a `configASSERT( pcHostName != NULL )`.

Reply:
I was just thinking from the perspective of CBMC checks. I would like to have proofs as unconstrained as possible. But if this configASSERT is a bit of an overstretch, then I am fine either way :)

Reply:
I will add the following:
~~~c
	/* Althought both pointers have been checked already, two extra
	asserts are added to help the CBMC proofs.. */
	configASSERT( pucUDPPayloadBuffer != NULL );
	configASSERT( pcHostName != NULL );
~~~